framework/src/packets/icmp/v6/ndp/options/link_layer_addr.rs

```diff
@@ -105,9 +135,67 @@ impl fmt::Display for LinkLayerAddress {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::packets::ethernet::Ethernet;
+    use crate::packets::icmp::v6::ndp::router_advert::RouterAdvertisement;
+    use crate::packets::icmp::v6::ndp::{SOURCE_LINK_LAYER_ADDR, TARGET_LINK_LAYER_ADDR};
+    use crate::packets::icmp::v6::Icmpv6;
+    use crate::packets::ip::v6::Ipv6;
+    use crate::packets::ip::ProtocolNumbers;
+    use crate::packets::{Packet, RawPacket};
+    use crate::testing::dpdk_test;
+    use std::str::FromStr;
 
     #[test]
     fn size_of_link_layer_address() {
         assert_eq!(8, LinkLayerAddressFields::size());
     }
+
+    #[dpdk_test]
+    fn test_add_source_link_layer_address() {
+        let mac_addr = MacAddr::from_str("01:00:00:00:00:00").unwrap();
+        let raw_packet = RawPacket::new().unwrap();
+        let eth = raw_packet.push::<Ethernet>().unwrap();
+        let mut ipv6 = eth.push::<Ipv6>().unwrap();
+        ipv6.set_next_header(ProtocolNumbers::Icmpv6);
+        let router_advert = ipv6.push::<Icmpv6<Ipv6, RouterAdvertisement>>().unwrap();
+
+        let mut option =
+            LinkLayerAddress::do_push_option(router_advert.mbuf(), router_advert.option_offset())
+                .unwrap();
+        option.set_addr(mac_addr);
+        option.set_option_type(SOURCE_LINK_LAYER_ADDR);
+
+        let source_link_layer_address =
+            LinkLayerAddress::parse(router_advert.mbuf, router_advert.option_offset()).unwrap();
+        assert_eq!(
+            source_link_layer_address.option_type(),
+            SOURCE_LINK_LAYER_ADDR
+        );
+        assert_eq!(source_link_layer_address.addr(), mac_addr);
```

Comment:
for these tests, I think we should exercise the public interface instead. use `options` and `push_option`.

Reply:
i removed these test from their respective option modules and put the tests inside router_advert since that's where we should call the public interface